### PR TITLE
DOMA-3379 fixed billing category and date picker locale in registry upload view

### PR DESCRIPTION
**Pull request by Alllex202**

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/common/components/Pickers/DatePicker.tsx [hunk 1]
```diff
@@ -1,7 +1,53 @@
...
-import generatePicker from 'antd/lib/date-picker/generatePicker'
+import generatePicker, {
+    PickerDateProps,
+    PickerProps,
+    PickerTimeProps,
+    RangePickerProps,
+} from 'antd/lib/date-picker/generatePicker'
+import { useIntl } from 'react-intl'
+import ruRU from 'antd/lib/locale/ru_RU'
+import enUS from 'antd/lib/locale/en_US'
```
Comment (pahaz) [inline apps/condo/domains/common/components/Pickers/DatePicker.tsx:12]: you need to move it to `index`

Comment (pahaz) [reply apps/condo/domains/common/components/Pickers/DatePicker.tsx:12]: It looks like we already have such code

Comment (pahaz) [reply apps/condo/domains/common/components/Pickers/DatePicker.tsx:12]: Please avoid to create a new places where you need to add a new language

**Review by dkoviazin — CHANGES_REQUESTED** [review]
You need to remove registry module update from your PR

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/common/components/Pickers/DatePicker.tsx (hunk 1 above)
Comment (Alllex202) [reply apps/condo/domains/common/components/Pickers/DatePicker.tsx]: fixed it, locale get from configProvider

File: apps/condo/domains/common/components/Pickers/DatePicker.tsx [hunk 2]
```diff
@@ -1,7 +1,45 @@
...
-export default DatePicker
\ No newline at end of file
+const generateDatepickerWithLocale = (Picker: PickerType) => (props) => {
+    const { locale } = useContext(ConfigProvider.ConfigContext)
+
+    return <Picker locale={locale.DatePicker} {...props} />
+}
+
+const MergedDatePicker = generateDatepickerWithLocale(DatePicker) as MergedDatePickerType
+MergedDatePicker.WeekPicker = generateDatepickerWithLocale(DatePicker.WeekPicker)
```
Comment (pahaz) [inline apps/condo/domains/common/components/Pickers/DatePicker.tsx:38]: You can’t call hooks outside of react component: https://reactjs.org/docs/hooks-rules.html

File: apps/condo/domains/common/components/Pickers/DatePicker.tsx [hunk 3]
```diff
@@ -1,7 +1,45 @@
...
+MergedDatePicker.MonthPicker = generateDatepickerWithLocale(DatePicker.MonthPicker)
+MergedDatePicker.YearPicker = generateDatepickerWithLocale(DatePicker.YearPicker)
+MergedDatePicker.RangePicker = generateDatepickerWithLocale(DatePicker.RangePicker)
+MergedDatePicker.TimePicker = generateDatepickerWithLocale(DatePicker.TimePicker)
+MergedDatePicker.QuarterPicker = generateDatepickerWithLocale(DatePicker.QuarterPicker)
+
+export default MergedDatePicker
```
Comment (pahaz) [inline apps/condo/domains/common/components/Pickers/DatePicker.tsx:45]: The name is looks strange.

Comment (sitozzz) [reply apps/condo/domains/common/components/Pickers/DatePicker.tsx:45]: @Alllex202 you may rename constant `DatePicker` to `DefaultDatePicker` (that created from result of `generatePicker(...)`) for example and then you be able to `export default DatePicker`

Comment (Alllex202) [reply apps/condo/domains/common/components/Pickers/DatePicker.tsx]: refactored variable names

**Pull request merged**